### DISCUSSION TITLE
Running immich with custom UID/gid

### DISCUSSION THREAD

**Discussion by whirlfire** [immich-app/immich#809] [Q&A]
Hello,

for security reasons, none of my containers run as root internally, but with a set UID/GUID.
In most containers, you can provide a UID/GUID via environment variables.

This provides security (a compromised container cannot install software inside the container, or outside if broken out) and also files and folders are created with those UID/GUID instead of root.

How can I achieve that? Any chance a global UID/GUID environment variable will be added for all containers?

**Comment by PixelJonas** [comment]
Hey, @Openhabber  there is currently an effort to make the containers run as `non-root` users. Once this is achieved, I guess moving towards any arbitrary users should be possible as well.

**Reply by alextran1502** [reply]
@Openhabber This PR https://github.com/immich-app/immich/pull/338 might give you some hints on how to do that

**Comment by petruut** [comment]
any update on this?

**Reply by mmomjian** [reply]
This is supported: https://immich.app/docs/FAQ#how-can-i-run-immich-as-a-non-root-user

**Comment by W3D3** [comment]
There is another image being built with UID/GUID environment variables that you can pass in, very useful for running Immich on UnRAID:
https://github.com/imagegenius/docker-immich?tab=readme-ov-file#docker-compose
